File: src/types/site.ts
export type SiteStatus = "running" | "stopped" | "starting" | "error";

export interface Site {
  id: string;
  name: string;
  domain: string;
  ps_version: string;
  php_version: string;
  mysql_version: string;
  port: number;
  pma_port: number;
  ssl_port: number | null;
  mail_port: number | null;
  status: SiteStatus;
  created_at: string;
}

export interface SiteTemplate {
  id: string;
  name: string;
  ps_version: string;
  php_version: string;
  mysql_version: string;
  is_default: number;
  user_created: number;
}

export interface CreateSiteInput {
  name: string;
  domain: string;
  ps_version: string;
  php_version: string;
  mysql_version: string;
}

File: src-tauri/src/models/site.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Site {
    pub id: String,
    pub name: String,
    pub domain: String,
    pub ps_version: String,
    pub php_version: String,
    pub mysql_version: String,
    pub port: i64,
    pub pma_port: i64,
    pub ssl_port: Option<i64>,
    pub status: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateSiteInput {
    pub name: String,
    pub domain: String,
    pub ps_version: String,
    pub php_version: String,
    pub mysql_version: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct SiteTemplate {
    pub id: String,
    pub name: String,
    pub ps_version: String,
    pub php_version: String,
    pub mysql_version: String,
    pub is_default: i64,
}
